server/data/data-cassandra/src/main/java/org/apache/james/rrt/cassandra/CassandraRecipientRewriteTable.java

```diff
@@ -98,14 +100,20 @@ protected Mappings mapAddress(String user, Domain domain) {
         Preconditions.checkArgument(listSourcesSupportedType.contains(mapping.getType()),
             "Not supported mapping of type %s", mapping.getType());
 
-        SchemaVersion schemaVersion = cassandraSchemaVersionDAO.getCurrentSchemaVersion()
-            .block()
-            .orElse(CassandraSchemaVersionManager.MIN_VERSION);
-
-        if (schemaVersion.isBefore(MAPPINGS_SOURCES_SUPPORTED_VERSION)) {
+        if (isLegacy()) {
             return super.listSources(mapping);
         }
 
         return cassandraMappingsSourcesDAO.retrieveSources(mapping).toStream();
     }
+
+    private boolean isLegacy() {
+        return isLegacy(initialSchemaVersion)
+            // If we started with a legacy james then maybe schema version had been updated since then
+            || isLegacy(versionManager.computeVersion());
```

Comment:
If we have started in legacy mode, the version will never be refreshed.

And it's not cached.

Reply:
Should be a &&

We miss testing.